examples/gpt-2/main.cpp

```diff
@@ -703,11 +796,16 @@ bool gpt2_eval(
     ggml_allocr_alloc_graph(allocr, gf);
 
     // run the computation
-    struct ggml_cplan plan = ggml_graph_plan(gf, n_threads);
-    static std::vector<uint8_t> work_buffer;
-    work_buffer.resize(plan.work_size);
-    plan.work_data = work_buffer.data();
-    ggml_graph_compute(gf, &plan);
+    if (ggml_backend_is_cpu(model.backend)) {
+        ggml_backend_cpu_set_n_threads(model.backend, n_threads);
+    }
+#ifdef GGML_USE_METAL
+    // TODO: not great - what should we do?
+    if (ggml_backend_is_metal(model.backend)) {
+        ggml_backend_metal_set_n_threads(model.backend, n_threads);
+    }
+#endif
```

Comment:
What is the issue here? There are always going to be some backend-specific functions, I think this is unavoidable.